api/lib/domain/usecases/add-pix-authentication-method-by-email.js

```diff
@@ -0,0 +1,11 @@
+const { AuthenticationMethodAlreadyExistsError } = require('../errors');
+
+module.exports = async function addPixAuthenticationMethodByEmail({ userId, authenticationMethodRepository }) {
+  const alreadyHasPixAuthenticationMethod = await authenticationMethodRepository.hasIdentityProviderPIX({ userId });
+
+  if (alreadyHasPixAuthenticationMethod) {
```

Comment:
Il était pas question à un moment de donner la responsabilité de la gestion des erreurs au repository ?
`hasIdentityProviderPIX` se prête pas bien car si je regarde le code, si `await authenticationMethodRepository.hasIdentityProviderPIX({ userId })` return true, ici c'est une erreur et dans le usecase `addPixAuthenticationMethodByEmail` ça déclenche autre chose.
@ajoanny tu aurais un conseil là dessus ?

Reply:
Actuellement, on a pas de pratiques d'équipe sur sur la gestions des erreurs dans les repo. La seule règle qu'on a (commune à toutes les équipes je crois) c'est que si une fonction / une variable commence par `has` / `is` ça doit renvoyer un booléen.
Perso je pense que si l'erreur est un cas "métier" c'est bien qu'elle apparaisse dans le usecase (dans le domain) plutôt que le repo (=infra). A mon sens les erreur repo sont des erreurs "techniques" qui ne devrait pas arriver dans un scénario classique... 🤔

Reply:
Merci ! Je vais voir si je retrouve la discussion sur le slack à ce propos